From 970a4b7a7d4338c08667ddab96eb92c252ca6d76 Mon Sep 17 00:00:00 2001
From: hoxnox <hoxnox@gmail.com>
Date: Mon, 11 Sep 2017 10:59:11 +0300
Subject: [PATCH 2/2] staticlibs

---
 cmake/FindFUSE.cmake | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/cmake/FindFUSE.cmake b/cmake/FindFUSE.cmake
index d854834..5da5e28 100644
--- a/cmake/FindFUSE.cmake
+++ b/cmake/FindFUSE.cmake
@@ -21,13 +21,22 @@ else (APPLE)
     set (FUSE_SUFFIXES fuse refuse)
 endif (APPLE)
 
+set( _fuse_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+if(WIN32)
+	set(CMAKE_FIND_LIBRARY_SUFFIXES .lib .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
+else()
+	set(CMAKE_FIND_LIBRARY_SUFFIXES .a )
+endif()
+
 # find includes
-find_path (FUSE_INCLUDE_DIR fuse.h
-        PATHS /opt /opt/local /usr/pkg
-        PATH_SUFFIXES ${FUSE_SUFFIXES})
+find_path (FUSE_INCLUDE_DIR fuse.h)
 
 # find lib
-find_library (FUSE_LIBRARIES NAMES ${FUSE_NAMES})
+find_library (FUSE_LIBRARIES NAMES fuse fuse3 PATH_SUFFIXES lib)
+if (FUSE_LIBRARIES_NOTFOUND AND NOT FUSE_USE_STATIC_LIBS)
+	set(CMAKE_FIND_LIBRARY_SUFFIXES ${_fuse_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
+	find_library (FUSE_LIBRARIES fuse PATH_SUFFIXES lib)
+endif()
 
 include ("FindPackageHandleStandardArgs")
 find_package_handle_standard_args ("FUSE" DEFAULT_MSG
-- 
2.13.0

